### ISSUE TITLE
TopicPartition should override equals() and hashCode()

### ISSUE THREAD

**Issue by daka1510** [vert-x3/vertx-kafka-client#96]
File:
- https://github.com/vert-x3/vertx-kafka-client/blob/master/src/main/java/io/vertx/kafka/client/common/TopicPartition.java

*Sample*
```
        // topic cleanup
        TopicPartition t1 = new TopicPartition("topic", 0);
        TopicPartition t2 = new TopicPartition("topic", 0);
        System.out.println(t1.equals(t2)); // yields false
```

**Comment by daka1510** [comment]
*Impact:*
Turned out to be an issue whenever instances of `TopicPartition` are used as a key within a Map or Set (which is part of the consumer APIs).

**Comment by vietj** [comment]
do you mind providing a patch ?

**Comment by daka1510** [comment]
@vietj - thanks for your prompt reaction. Can you review the attached PR?

**Comment by ppatierno** [comment]
Addressed by #97
